### ISSUE TITLE
Should add current URL hash to clientData

### ISSUE THREAD

**Issue by wch** [rstudio/shiny#1418]
Currently, `clientData` only contains the initial URL hash, but it doesn't have the current URL hash. This is because not all browsers at the time supported the `onhashchange` event. See the comment here:
  https://github.com/rstudio/shiny/blob/9613c58/srcjs/init_shiny.js#L412-L416

Pretty much all browsers these days support it, even IE8. See:
  http://caniuse.com/#search=onhashchange
  http://benalman.com/projects/jquery-hashchange-plugin/

**Comment by daattali** [comment]
I'm used to the assumption that `session$clientData` only runs once for each session, when the session begins. Would this be the only case where clientData can get updated during a session? Or was my previous assumption wrong

**Comment by wch** [comment]
`session$clientData` is a `reactiveValues` object. Some of the things don't change, but there are things that do. See here: https://github.com/rstudio/shiny/blob/7aacf9c/R/shiny.R#L206-L212

**Comment by bborgesr** [comment]
@wch this is about to be merged in the pseudo-nav PR (#1447). Here's the sample app (using the `barbara@pushstate` branch) -- also in documentation, see `?getUrlHash`:

```r
  ## Printing the value of the URL hash
  ## (Use the back and forward buttons to see how the browser
  ## keeps a record of each state)
  shinyApp(
    ui = fluidPage(
      textInput("txt", "Enter new hash"),
      helpText("Format: #hash"),
      actionButton("go", "Update"),
      hr(),
      verbatimTextOutput("hash")
    ),
    server = function(input, output, session) {
      observeEvent(input$go, {
        updateQueryString(input$txt, mode = "push")
      })
      output$hash <- renderText({
        hash <- getUrlHash()
        paste("Your hash is:\n", hash)
      })
    }
  )
```

What do you think?

**Comment by bborgesr** [comment]
Closed in #1447 via  a0b917a
